### ISSUE TITLE
trouble compiling

### ISSUE THREAD

**Issue by coachshea** [volftomas/gbook#1]
I have recently discovered your project and it appears to be exactly what I have been looking for. whenever I try to compile it though, I receive the following error:

-- Found JSONCpp: /usr/lib/libjsoncpp.so
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
GMOCK_LIBRARY
    linked by target "test_it" in directory .../gbook
GTEST_LIBRARY
    linked by target "test_it" in directory .../gbook

-- Configuring incomplete, errors occurred!
See also ".../gbook/CMakeFiles/CMakeOutput.log".

**Comment by volftomas** [comment]
The issue seems to be related to building of tests. I'll take a look today (I hope). In the mean time (before proper fix), can you try to comment out/delete lines 69 - 91 in CMakeLists.txt? You should be able to build it.

**Comment by volftomas** [comment]
@coachshea ok, and of course I recommend doing backup before trying it out, the fact the it works for me doesn't mean it will for you :)

**Comment by coachshea** [comment]
When I comment out those lines and try to run cmake, I get the following 
errors:


-- Found JSONCpp: /usr/lib/libjsoncpp.so
-- Configuring done
CMake Error at CMakeLists.txt:56 (ADD_EXECUTABLE):
  Cannot find source file:

    lib/tinyxml2/tinyxml2.cpp

  Tried extensions .c .C .c++ .cc .cpp .cxx .m .M .mm .h .hh .h++ .hm .hpp
  .hxx .in .txx

**Comment by volftomas** [comment]
I would guess you didn't check out lib/tinyxml2 git submodule. I should probably put that into readme.

For git 1.6.5 and later run in root of the repository:

git submodule update --init --recursive

it's from http://stackoverflow.com/a/4438292/781743 , I don't have access to linux machine at the moment to verify it myself


PS: you are first person I know of, besides me, to try to compile this, so I'm sorry for the bumpy road :/

**Comment by coachshea** [comment]
That was it. It compiles fine now. I will check it out as soon as I can and 
I'll let you know how it goes.

Thanks for your help.

**Comment by coachshea** [comment]
When I use it I get an endless loop of:
} in /home/coachshea/ignore/gbook/src/oauth2.cpp:begin_polling:71
07:59:52:DEBUG2 : oauth2         : No tokens yet, waiting some more in /home/coachshea/ignore/gbook/src/oauth2.cpp:begin_polling:88
07:59:57:DEBUG2 : oauth2         : Looping... in /home/coachshea/ignore/gbook/src/oauth2.cpp:begin_polling:61
07:59:58:DEBUG2 : oauth2         : Response received. in /home/coachshea/ignore/gbook/src/oauth2.cpp:begin_polling:70
07:59:58:DEBUG3 : oauth2         : {
  "error" : "authorization_pending"

I'm not sure if this is in regards to gbook or my google credentials. I have never used client_id and client_secret before, so that could be the case, but I tried to set it up step by step from internet tutorials and I'm not sure what I got wrong, if that is in fact where the problem is.

**Comment by volftomas** [comment]
Hi :) So I took a look at it. I updated the build system and readme. Tests are now not build by default, so you can just pull and overwrite your changes, let me know if it still builds.

In regards to your new problem: first, make sure that in your Google Developers Console, the credential for gbook (the client_id & secret_id) is of type "Other". I'm not sure it matters, but "Other" works for me.

When you run gbook for first time, it should prompt you (into stdout) with something like following:

    To continue please go to:
        SOME_URL
    and enter the following code when prompted to:
        SOME_CODE

You need to go to the URL in a browser and enter the code. That's the way Google (and oauth2) chose to verify access for console applications. The thing you see in the log (begin_polling ...) is waiting before you authorize gbook access on the URL with the code. This should be remember however, so you don't have to do that every time (at least I hope :D).

Let me know if I was of any help.

**Comment by coachshea** [comment]
You were right. I had to set type to "Other". In case anyone else comes across 
this, I also had to explicitly enable the contacts api on google, and it now 
works perfectly.

Thank you so much for all your help and thank you for this great product. It 
has been a dream of mine for a long time to be able to use abook on my 
computer without the need to keep two sets of contacts in sync manually.

Thanks again.
